howtos/module/_posts/2016-06-30-Ten-ways-improve-modules.md

```diff
@@ -0,0 +1,110 @@
+Did you know that it is possible to extend these templates and that you can define your own configuration elements? Here's a code snippet that will allow you to use PrestaShop 1.6's switch element on an older version of PrestaShop:
+
+{% gist firstred/b5016aab980ade678d741691e5b15c5b %}
+
+You can simply place this file in the location `views/templates/admin/_configure/helper/form/form.tpl` relative to the root of your module's folder and the `HelperForm` template has been extended.
+Do you see the `$smarty.const._PS_VERSION_|@addcslashes:'\'' < '1.6'` check? This simply allows us to define several templates for different versions of PrestaShop. Pretty cool if you want to keep your code together and the view separated in a different file.
+It also keeps your main module file readable as you will only need to define an array with the right values in order to generate the form. Remember PrestaShop's MVC nature? You're perfectly respecting this pattern by implementing your configuration page this way.
+
+### 4. Contribute to PrestaShop's core code
+PrestaShop's core code is baked on [GitHub](https://github.com/PrestaShop/PrestaShop). This is where developers gather and contribute to PrestaShop. Not only do developers contribute code, you will also see some of the visitors comment on your Pull Request.
+Before your contribution is merged, it will also be checked by PrestaShop's developers and quality assurance team. Sometimes they might merge your code right away, but it can also happen that your code contains some bugs.
+The advantage of such an open and transparent process, is that others will see your code (guaranteed, as all pull requests are at least being checked once) and might reveal some bad coding habits you might have developed over time, without you even realizing it.
+Most contributors visit the GitHub repository in order to improve PrestaShop and the contributions, so the chance of someone suggesting improvements for your code is very high. Use the advice to your advantage!
+
+### 5. Drop support for older versions
+As PrestaShop's code improves over time, so does the API. The newer the version, the easier it will be to develop a module as PrestaShop's codebase has evolved over time. It is now much easier to create a module than for example four years ago.
+By dropping support for older version you will not only enjoy the features of a newer API, you will also have to insert less conditional blocks into your code. In the end, this makes it a lot easier to test and maintain your code.
+
+### 6. Use open source modules as "boilerplates"
+PrestaShop 1.6's developer documentation, unfortunately, is far from finished as there is enough left to write about. You might have had numerous time where you had wished that the page contained some more information.
+Luckily, most of the basics can be figured out by having a look at the basic modules on GitHub or other open source PrestaShop module projects. Yes, they exist :)
```

Comment:
It's hard to find good open source modules. Do you happen to know a few?

I at least remember this one: https://github.com/dim00z/gshoppingflux